pkg/process/checks/net.go

```diff
@@ -112,13 +115,38 @@ func (c *ConnectionsCheck) enrichConnections(conns []*model.Connection) []*model
 	return conns
 }
 
+func (c *ConnectionsCheck) diffTelemetry(tel *model.ConnectionsTelemetry) *model.CollectorConnectionsTelemetry {
+	if tel == nil {
+		return nil
+	}
+
+	cct := &model.CollectorConnectionsTelemetry{
+		KprobesTriggered:          tel.MonotonicKprobesTriggered - c.lastTelemetry.KprobesTriggered,
+		KprobesMissed:             tel.MonotonicKprobesMissed - c.lastTelemetry.KprobesMissed,
+		ConntrackRegisters:        tel.MonotonicConntrackRegisters - c.lastTelemetry.ConntrackRegisters,
+		ConntrackRegistersDropped: tel.MonotonicConntrackRegistersDropped - c.lastTelemetry.ConntrackRegistersDropped,
+		DnsPacketsProcessed:       tel.MonotonicDnsPacketsProcessed - c.lastTelemetry.DnsPacketsProcessed,
+		ConnsClosed:               tel.MonotonicConnsClosed - c.lastTelemetry.ConnsClosed,
+		ConnsBpfMapSize:           tel.ConnsBpfMapSize,
+	}
```

Comment:
Is there an edge case where we report the monotonic counts on the first check here, or are we dropping the data for the first check somewhere?

If the `process-agent` is restarted and connects to an existing system-probe, then `c.lastTelemetry` will be filled with zero values and we'll end up setting these fields to `tel.Monotonic* - 0`, no?

Reply:
You are correct that if `process-agent` restarts, we would report the full monotonic values. I can add something to drop the first value.